pluggy/callers.py

```diff
@@ -0,0 +1,108 @@
+'''
+Call loop machinery
+'''
+import sys
+
+
+_py3 = sys.version_info > (3, 0)
+
+
+if not _py3:
+    exec("""
+def _reraise(cls, val, tb):
+    raise cls, val, tb
+""")
+
+
+def _raise_wrapfail(wrap_controller, msg):
+    co = wrap_controller.gi_code
+    raise RuntimeError("wrap_controller at %r %s:%d %s" %
+                       (co.co_name, co.co_filename, co.co_firstlineno, msg))
+
+
+class HookCallError(Exception):
+    """ Hook was called wrongly. """
+
+
+class Result(object):
```

Comment:
Yeah I really didn't like the duplication.
@RonnyPfannschmidt are you ok with keeping `_Result` and being rid of `_CallOutcome`?